CalcLibCore/Mori/Calculator.cs

```diff
@@ -0,0 +1,314 @@
+            }
+            // クリアボタン
+            else if (btn == CalcButton.BtnClear)
+            {
+                context.Reset();
+            }
+        }
+    }
+
+    /// <summary>
+    /// イコール入力状態
+    /// </summary>
+    internal class EqualState : IState
+    {
+        private static IState singleton = new EqualState();
+        private EqualState() { }
+        public static IState GetInstance()
+        {
+            return singleton;
+        }
+
+        public void AcceptInput(CalcContextExtend context, CalcButton btn)
+        {
+            // 数字ボタンの場合は数字入力状態に戻す
+            if (btn >= CalcButton.Btn0 && btn <= CalcButton.Btn9)
+            {
+                context.Reset();
+                context.State.AcceptInput(context, btn);
+            }
+            // イコールのあとに演算子がくると計算結果が左辺になり演算子ストラテジーを変更する
+            else if (btn >= CalcButton.BtnPlus && btn <= CalcButton.BtnMultiple)
+            {
+                context.Operand = context.CurrentValue;
+                context.Strategy = context.ChangeStrategy(btn); // 演算子を変更する
+                context.SubDisplayText = $"{context.CurrentValue} {context.GetOperatorString(btn)}";
+                context.State = OperatorState.GetInstance(); // 演算子入力状態に遷移
+                context.isNewNumberInputStarted = true; // 新しい数字入力を受け付けられるようにする
+            }
+            // イコールボタンは同じ計算を再実行する
+            else if (btn == CalcButton.BtnEqual)
+            {
+                var leftOperand = context.CurrentValue;
+                context.CurrentValue = context.Strategy.Execute(context.CurrentValue, context.Operand);
+                context.DisplayText = context.CurrentValue.ToString();
+                context.SubDisplayText = $"{leftOperand} {context.GetOperatorString(null)} {context.Operand} =";
+            }
+            // クリアボタン
+            else if (btn == CalcButton.BtnClear)
+            {
+                context.Reset();
+            }
+        }
+    }
+
+
+    // 計算のロジックを管理するクラス
+    internal interface ICalculationStrategy
+    {
+        decimal Execute(decimal operand1, decimal operand2);
+    }
+
+    // 足し算
+    internal class AdditionStrategy : ICalculationStrategy
+    {
+        public decimal Execute(decimal operand1, decimal operand2)
+        {
+            return operand1 + operand2;
+        }
+    }
+
+    // 引き算
+    internal class SubtractionStrategy : ICalculationStrategy
+    {
+        public decimal Execute(decimal operand1, decimal operand2)
+        {
+            return operand1 - operand2;
+        }
+    }
+
+    // 掛け算
+    internal class MultiplicationStrategy : ICalculationStrategy
+    {
+        public decimal Execute(decimal operand1, decimal operand2)
+        {
+            return operand1 * operand2;
+        }
+    }
+
+    // 割り算
+    internal class DivisionStrategy : ICalculationStrategy
+    {
+        public decimal Execute(decimal operand1, decimal operand2)
+        {
+            if (operand2 == 0) return 0;
```

Comment:
`0` は int 型のリテラルなので、decimalのリテラルをかくべきだと思うので、末尾に `m` をつけておくのが正解ですね。
```suggestion
            if (operand2 == 0m) return 0m;
```